src/System.ComponentModel.TypeConverter/src/System/ComponentModel/InterlockedBitVector32.cs

```diff
@@ -0,0 +1,59 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Threading;
+
+namespace System.ComponentModel
+{
+    /// <summary>
+    /// Provides a subset of the <see cref="BitVector32"/> surface area, using volatile operations for reads and interlocked operations for writes. 
+    /// </summary>
+    internal struct InterlockedBitVector32
+    {
+        private int _data;
+
+        public InterlockedBitVector32(int data)
+        {
+            _data = data;
+        }
+
+        public bool this[int bit]
+        {
+            get { return (Volatile.Read(ref _data) & bit) == bit; }
+            set
+            {
+                while (true)
+                {
+                    int oldValue = _data;
+                    int newValue = value ? oldValue | bit : oldValue &= ~bit;
```

Comment:
Current call sites for the indexer and DangerousSet all pass `true` values. Is the ability to turn off bits needed?